src/celldega/nbhd/hextile.py

```diff
@@ -38,9 +42,51 @@ def generate_hex_grid(
                 hexagons.append(hex_tile)
 
     return gpd.GeoDataFrame(
-        {
-            "name": [f"hex_{i}" for i in range(len(hexagons))],
-            "geometry": hexagons,
-        },
+        {"name": [f"hex_{i}" for i in range(len(hexagons))], "geometry": hexagons},
         crs=gdf_cell.crs,
     )
+
+
+def prepare_nbhd_for_clustering(
+    adata: AnnData,
+    nbhd_type: str,
+    radius: float = 75,
+    path_landscape_files: str = None
+) -> tuple[gpd.GeoDataFrame, gpd.GeoDataFrame]:
+    """Prepare nbhd geometries for clustering based on cell type composition.
+    Also returns cell-type proportions in each nbhd."""
+
+    if not isinstance(adata, AnnData):
+        raise TypeError("adata must be an AnnData object")
+
+    if nbhd_type == 'HEX':
+        gdf_object = _get_gdf_cell(adata)
+        gdf_nbhd = generate_hex_grid(gdf_object, radius=radius)
+
+    elif nbhd_type == 'ALPH':
+        alphas_list = range(30, 60, 5)
+        gdf_object = alpha_shape_cell_clusters(adata, cat='cluster', alphas=alphas_list)
+        gdf_nbhd = gdf_object.loc[gdf_object['inv_alpha']==35]
+
+    elif nbhd_type == 'SKTCH':
+        gdf_nbhd = gpd.read_parquet(f'{path_landscape_files}/spatial_regions/sketched_regions_micron.parquet')
+        gdf_nbhd['name'] = gdf_nbhd['roi']
+
+    elif nbhd_type == 'GRAD':
+        gdf_object = _get_gdf_cell(adata)
+        gdf_micron = gpd.read_parquet(f'{path_landscape_files}/spatial_regions/sketched_regions_micron.parquet')
+        gdf_roi = gdf_micron.loc[gdf_micron['roi'] == 'region_1']
+        gdf_nbhd = calc_grad_nbhd_from_roi(gdf_roi, gdf_object, 200)
+        gdf_nbhd['name'] = gdf_nbhd['band']
+
+    # Calculate cell-type counts per nbhd tile
+    counts, population_distribution = calc_nbp(gdf_object, gdf_nbhd)
+
+    # Prepare AnnData for clustering
+    ad_tiles = AnnData(
+        X=population_distribution.values,
+        obs=pd.DataFrame(index=population_distribution.index),
+        var=pd.DataFrame(index=population_distribution.columns),
+    )
+
+    return ad_tiles, gdf_nbhd
```

Comment:
lets get rid of this function and this will simplify the outputs so that we do not have to pass back the gdf_nbhd to the user since this will be an argument instead.